### PR TITLE
Add a grand total event counter to the status output

### PR DESCRIPTION
**Pull request by 6yozo**
Add a grand total event counter to the status output

### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Review this pull request file-by-file.

### CODE REVIEW

File: libvast/src/system/index.cpp [hunk 1]
```diff
@@ -634,11 +634,16 @@ index_state::status(status_verbosity v) const {
...
+    auto sum = uint64_t{0};
     for (const auto& [name, layout_stats] : stats.layouts) {
       auto xs = record{};
       xs["count"] = count{layout_stats.count};
       layout_object[name] = xs;
+      sum += layout_stats.count;
     }
+    auto xs = record{};
+    xs["count"] = count{sum};
+    layout_object["total"] = xs;
```
Comment (dominiklohmann) [inline libvast/src/system/index.cpp:646]: Doesn't this override statistics if we ever have a schema named `total`?

**Review by dominiklohmann — COMMENTED** [review]
Some minor remarks:
- The first two commits in this look unrelated to me.
- There's an integration test that needs an adjustment.
- Right now there's an ambiguity if there's a schema named `total`, which I think we can avoid by putting the sum in the outer statistics object.

File: changelog/unreleased/features/2133--status-grand-total.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The output of `vast status --detailed` now displays the total number of events
+stored in the DB.
```
Comment (dominiklohmann) [inline changelog/unreleased/features/2133--status-grand-total.md:2]: As a user the first question I have when reading this is: Under what key?

File: vast/integration/reference/server-zeek-conn-log/step_07.ref [hunk 3]
```diff
@@ -16,6 +16,7 @@
 {"path":["index","num-cached-partitions"],"type":"number"}
 {"path":["index","num-unpersisted-partitions"],"type":"number"}
 {"path":["index","partitions","active",0,"id"],"type":"string"}
+{"path":["index","statistics","all-layouts","total"],"type":"number"}
```
Comment (tobim) [inline vast/integration/reference/server-zeek-conn-log/step_07.ref:19]: Imho the extra `all-layouts` key doesn't really add value, how about just:
```suggestion
{"path":["index","statistics","events"],"type":"number"}
```

File: changelog/unreleased/features/2133--status-grand-total.md [hunk 4]
```diff
@@ -0,0 +1,2 @@
+The output of `vast status --detailed` now displays the total number of events
```
Comment (tobim) [inline changelog/unreleased/features/2133--status-grand-total.md:1]: I think that information is important enough to be displayed on the `info` level.

**Review by tobim — CHANGES_REQUESTED** [review]
I think we should emit this info for `vast status` without `--detailed` too.

File: vast/integration/reference/server-zeek-conn-log/step_07.ref (hunk 3 above)
Comment (dominiklohmann) [reply vast/integration/reference/server-zeek-conn-log/step_07.ref:19]: We could also switch the structure more generally, e.g.,

```json
"events": {
  "total": 123,
  "schema": {
    "suricata": {
      "alert": 1,
      "flow": 19
    },
    "zeek": {
      "conn": 100,
      "http": 3
    }
  }
}
```

Comment (6yozo) [reply vast/integration/reference/server-zeek-conn-log/step_07.ref:19]: So the new output of `vast status --detailed` should be like:
```yaml
{
  ...
  "index": {
    ...
    "events": {
      "total": 25711
       schema": {
         "suricata": {
           "alert": 1,
           "flow": 19
       },      
       ...     
    },
    ...
  },
  ...
}
```

Comment (6yozo) [reply vast/integration/reference/server-zeek-conn-log/step_07.ref:19]: Do we achieve that format by splitting the `name` in `stats.layouts` by dots?

https://github.com/tenzir/vast/blob/622a00dc8c5501c97bc8df8e5e9fdeb1deb6674f/libvast/src/system/index.cpp#L638

Comment (dominiklohmann) [reply vast/integration/reference/server-zeek-conn-log/step_07.ref:19]: It's just an idea for an improved UX. Sounds like a good thing to discuss in Slack quickly based on a list of possible outputs. We need to break here anyways given that we want to `s/schema/module` and `s/layout/schema`, and while we're at it we should also think about the ergonomics.

**Review by dominiklohmann — APPROVED** [review]

File: libvast/src/system/index.cpp [hunk 5]
```diff
@@ -631,19 +631,22 @@ index_state::status(status_verbosity v) const {
...
   auto rs = make_status_request_state<extra_state>(self);
+  auto stats_object = record{};
+  auto sum = uint64_t{0};
+  for (const auto& [_, layout_stats] : stats.layouts)
+    sum += layout_stats.count;
+  auto xs = record{};
+  xs["total"] = count{sum};
+  stats_object["events"] = xs;
+  if (v == status_verbosity::info)
+    rs->content["statistics"] = std::move(stats_object);
```
Comment (tobim) [inline libvast/src/system/index.cpp:642]: nit: We could also do this assignment unconditionally at the end (before the `fill_status_map` stuff). I don't have a strong opinion though so you are free to ignore this suggestion.

**Review by tobim — APPROVED** [review]

**Pull request merged**